Appraisals

```diff
@@ -11,10 +19,27 @@ end
 
 alias original_appraise appraise
 
+REMOVED_GEMS = {
+  :check => [
+    'rbs',
+    'steep',
+  ],
+}
+
 def appraise(group, &block)
   # Specify the environment variable APPRAISAL_GROUP to load only a specific appraisal group.
   if ENV['APPRAISAL_GROUP'].nil? || ENV['APPRAISAL_GROUP'] == group
-    original_appraise(group, &block)
+    original_appraise(group) do
+      instance_exec(&block)
+
+      REMOVED_GEMS.each do |group_name, gems|
+        group(group_name) do
+          gems.each do |gem_name|
+            remove_gem gem_name if respond_to?(:remove_gem)
```

Comment:
There doesn't seem to be any Ruby version where `respond_to?(:remove_gem)` is `false` -- grepping on the `gemfiles/` that you generated, these two gems seemed to have been removed on every one of them.

Thus, I would suggest removing the `if respond_to?(:remove_gem)`. This is code only used for development/CI, so it's best that it fails loudly if it behaves unexpectedly, rather than continue to chug along silently when it's not doing what we thought it should do.

Reply:
> grepping on the gemfiles/ that you generated, these two gems seemed to have been removed on every one of them.

Your test is good but you used it as a proxy that led you to an incorrect conclusion :)

2.1 and 2.2 can only use `appraisal` 2.2, and `remove_gem` appeared in `appraisal` 2.3.

It just so happens that it's okay because the removed gems are `rbs` and `steep` which are conditioned out by a version check in the main Gemfile.

By the way, I forgot to add `sorbet` and `spoom` in there. I wish there were a `remove_group` method!

Now, why use `remove_gem` and not just `without check`? Because bundler is descriptive and would nonetheless add `rbs` and `steep` to the lockfile so that they are still pinned to resolved versions at time of lockfile generation should one use the lockfile with `:check` on another system.

Reply:
Ah, thanks for the clarification. Can I trouble you to add a note that that line is only for Ruby 2.1 or 2.2? That way we can grep the codebase when we drop them in the future and then get rid of this as well :)